WordPressAuthenticator/Authenticator/WordPressAuthenticator.swift

```diff
@@ -177,6 +177,10 @@ import WordPressKit
             return nil
         }
 
+        if let loginNavController = controller as? LoginNavigationController, let loginPrologueViewController = loginNavController.viewControllers.first as? LoginPrologueViewController {
```

Comment:
The `restrictToWPCom` and `onLoginButtonTapped` remains to be unused, I wonder if something else is broken because of this 🤔

There used to be such code and then removed in this PR https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/640: 

```
if let childController = controller.children.first as? LoginPrologueViewController {
        childController.loginFields.restrictToWPCom = restrictToWPCom
        childController.showCancel = showCancel
        childController.onLoginButtonTapped = onLoginButtonTapped
}
```

Maybe such solution needs to be restored?

Reply:
Thanks for looking!  I wouldn't know how to update the PR steps to test these changes and we have been changing login flow because of Jetpack recently.  It might be worth exploring but not as part of this PR I think

Reply:
@mokagio just wanted to tag you in here.  It seems like I'm adding back functionality that was removed as part of this change [here](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/640/files#diff-8b77840eb74486edc937048220e6c0c491d2789095582e5a3029c1406a248e20L179).  Would you mind adding some context on why this might have been removed as part of removing Lottie?  Thanks!

Reply:
@sla8c 

> Would you mind adding some context on why this might have been removed as part of removing Lottie? 

I'm mostly sure that was an oversight. Or rather, a case of not having enough automation to know when removing code  that looks dead from the library point of view (like in the case of [that check](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/640/files#diff-8b77840eb74486edc937048220e6c0c491d2789095582e5a3029c1406a248e20L179)) might result in unexpected behavior downstream.

For additional context, I looked at the Git history and found out that:

1. The whole of `LoginPrologueViewController` was removed [in the first phase of removing Lottie](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/640/commits/89bb95976566faa0baf291e842f7cc733d840698)
2. That, of course, resulted in that check for the first child view controller being a `LoginPrologueViewController` being deleted as well
3. While the library built that way, the Woo UI tests failed, which prompted me to [reintroduce the type](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/640/commits/cf675a96481a79ce6413e1780e89af9c5dd96bbe)
4. It didn't occur to me to reintroduce the check as well, and because there was no test that failed, the whole thing went under the radar until you started working on it.

I don't remember the rational for removing `LoginPrologueViewController` in the first place, but my guess would be that it was thought to be part of a Lottie-driven set of UI components that should no longer have been used. Just a guess, though, based on the other deleted files. I could be off the mark and haven't dug deeper in the code and history.

@sla8c @staskus from the comments above, it looks like you are already on top of this, but I'd stress that it would be very useful to have an automated test to ensure the behavior restored with this PR doesn't degenerate again in the future. I guess given the issue was in the Jetpack/WordPress codebase, the test could live there, at least until we get to adding a demo app to this project. I'm sure @wordpress-mobile/mobile-ui-testing-squad will be happy to help with that.

Reply:
@mokagio thanks for the context!  @wordpress-mobile/mobile-ui-testing-squad please let me know if you need anything (a GH issue created) or anything else re: getting a UI test for this on your radar thanks 🙂

Reply:
@sla8c if you have the time, it would be super helpful if you could have a crack at writing the test. 

The infrastructure @wordpress-mobile/mobile-ui-testing-squad is building is meant to empower developers to write tests across the entire low-level/unit-test to high-level/UI-test spectrum. It's always helpful to get feedback on the state of the tooling and the codebase itself from the "users" we're building things for.